### ISSUE TITLE
wrong timezone in Ireland

### ISSUE THREAD

**Issue by shardtor** [dateutil/dateutil#1009]
`tzlocal()` gives the wrong timezone in Ireland. Ireland is currently on GMT not IST:

    $ cat timeissue.py 
    #!/usr/bin/env python
    from datetime import datetime
    from dateutil.tz import tzlocal, gettz

    now1 = datetime.now(tzlocal())
    print now1, now1.tzname()

    now2 = datetime.now(gettz())
    print now2, now2.tzname()
    $ ./timeissue.py 
    2020-02-29 19:54:56.407405+01:00 IST
    2020-02-29 18:54:56.411176+00:00 GMT
    $ date
    Sat 29 Feb 2020 06:56:55 PM GMT
    $ python --version
    Python 2.7.16
    $ python -c 'import dateutil; print(dateutil.__version__)'
    2.8.1

**Comment by pganssle** [comment]
Hm.. I don't have time at the moment to figure this out, but I'm fairly sure it's something to do with the libraries the `time` module is implemented in terms of doing the wrong thing with regards to Ireland's negative DST, because it seems that:

1. `time.timezone` is 0 and `time.altzone` is -3600, same as in London.
2. The order for `time.tzname` is `(GMT, IST)` not `(IST, GMT)` as I'd expect.
3. `time.localtime(datetime.now())` correctly returns `tm_isdst=1`.

I think this is actually a bug in Python's implementation of `time.timezone`, `time.altzone` and `time.tzname` because [the documentation](https://docs.python.org/3/library/time.html#timezone-constants) says that `altzone` is "the offset of the local DST timezone", but in this case it's returning the offset of the *standard* time zone.

Nothing can be done for Python 2, though it can be fixed for Python 3; the correct workaround (and probably for the best in general) is to use `tz.gettz()`, which should give you a `tzfile()` if your `TZ` environment is set correctly.

I'd be happy to introduce a workaround in `dateutil` if there's some way to determine when to reverse the order of `timezone` and `altzone` that doesn't hit this bug (and that won't get broken if this bug is fixed), but I can't think of one offhand.

**Comment by shardtor** [comment]
It seems the issue isn't in Python. tm_dst should be 0:
```
$ cat timeissue.c
#include <time.h>
#include <stdio.h>

int main()
{
    struct tm * myTime;
    time_t theTime;
    time(&theTime);
    myTime = localtime ( &theTime );
    printf("%s",asctime(myTime));
    printf("  tm_isdst = %d\n",myTime->tm_isdst);
}
$ TZ=Europe/Dublin ./timeissue 
Sun Mar  1 11:46:15 2020
  tm_isdst = 1
$ TZ=Europe/London ./timeissue 
Sun Mar  1 11:46:19 2020
  tm_isdst = 0
```

**Comment by pganssle** [comment]
No, `tm_idst` should be 1, I've filed a bug in Python: https://bugs.python.org/issue39804

This is because Ireland uses [Winter Time](https://en.wikipedia.org/wiki/Winter_time_(clock_lag)). That is why `datetime.now(tz.gettz()).dst()` returns a non-zero (negative) value in Ireland right now.

**Comment by shardtor** [comment]
Thanks. I see, GMT should be the DST zone.

**Comment by phord** [comment]
Isn't this a dup of https://github.com/regebro/tzlocal/issues/80 ?
